xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/transactions/MpTokenAmount.java

```diff
@@ -0,0 +1,71 @@
+package org.xrpl.xrpl4j.model.transactions;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.primitives.UnsignedLong;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Auxiliary;
+import org.immutables.value.Value.Derived;
+import org.immutables.value.Value.Immutable;
+
+/**
+ * {@link CurrencyAmount} type for MPT amounts.
+ */
+@Immutable
+@JsonSerialize(as = ImmutableMpTokenAmount.class)
+@JsonDeserialize(as = ImmutableMpTokenAmount.class)
+public interface MpTokenAmount extends CurrencyAmount {
```

Comment:
Minor nit -- any objection to renaming this to `MpTokenCurrencyAmount` since it extends `CurrencyAmount`?

This suggestion feels correct from a Java-perspective (i.e., class-name should mirror the interface name) although admittedly, the suggestion feels a bit askew from an MPT perspective, because MPTs support more than just "Currency".

Thoughts @nkramer44 ?